From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sat, 18 Apr 2026 23:06:36 -0500
Subject: [PATCH 17/46] 
 0022-winewayland-Avoid-constantly-reconfiguring-the-clien.patch

---
 dlls/winewayland.drv/wayland_surface.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index 2586a4aa2bd..bcc94ce6474 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -1339,10 +1339,13 @@ static void wayland_client_surface_update(struct client_surface *client)
 
     if (!(data = wayland_win_data_get(hwnd))) return;
 
-    if (toplevel && NtUserIsWindowVisible(hwnd))
-        wayland_client_surface_attach(surface, toplevel);
-    else
-        wayland_client_surface_attach(surface, NULL);
+    if (data->client_surface != surface)
+    {
+        if (toplevel && NtUserIsWindowVisible(hwnd))
+            wayland_client_surface_attach(surface, toplevel);
+        else
+            wayland_client_surface_attach(surface, NULL);
+    }
 
     wayland_win_data_release(data);
 }
